packages/codec/lib/interface/decoders/contract.ts

```diff
@@ -382,10 +477,21 @@ export class ContractInstanceDecoder {
     return await this.wireDecoder.getCode(address, block);
   }
 
-  //EXAMPLE: to watch a.b.c[d][e], use watchMappingKey("a", "b", "c", d, e)
-  //(this will watch all ancestors too, or at least ones given by mapping keys)
-  //feel free to mix arrays, mappings, and structs here!
-  //see the comment on constructSlot for more detail on what forms are accepted
+  /**
+   * Watches a mapping key; adds it to the decoder's list of watched mapping keys.
+   * This affects the results of both variables() and variable().
+   * When a mapping is decoded, only the values at its watched keys will be included in its value.
+   * Note that it is possible to watch mappings that are inside structs, arrays, other mappings, etc;
+   * see below for more on how to do this.
+   * Note that watching mapping keys is only possible in full mode; if the decoder wasn't able to
+   * start up in full mode, this method will throw an exception.
+   * Warning: At the moment, this function does very little to check its input.  Bad input may have
+   * unpredictable results.  This will be remedied in the future (by having it throw exceptions on
+   * bad input), but right now essentially no checking is implemented.
+   * Also, there may be slight changes to the format of indices in the future.
+   * @param variable The variable that the mapping lives under; this works like the nameOrId argument to variable().  If the mapping is a top-level state variable, put the mapping itself here.  Otherwise, put the top-level state variable it lives under.
+   * @param indices Further arguments to watchMappingKey, if given, will be interpreted as indices into or members of the variable identified by the variable argument.  Thus, if we have a struct type MapStruct with a mapping(string => string) called "map", and we have a variable arr of type MapStruct[], then one could watch arr[3].map["hello"] by calling watchMappingKey("arr", 3, "map", "hello").  Array indices and mapping keys are specified by value; struct members are specified by name or (if you know it) numeric ID.  Numeric values can be given as number, BN, or numeric string.  Bytestring values are given as hex strings.  Boolean values are given as booleans, or as the strings "true" or "false".  Address values are given as hex strings; they are currently not required to be in checksum case, but this will likely change in the future, so don't rely on that.  Note that if the path to a given mapping key includes mapping keys above it, any ancestors will also be watched automatically.  E.g., if m is of type mapping(uint => mapping(uint => uint)), then watching m[3][5] will also automatically watch m[3]; otherwise, watching m[3][5] wouldn't do much of anything.
```

Comment:
Can your example here be given as a doc comment `@example` (or whatever construct is available)?

Or at least split out so it's not one long paragraph?

Reply:
(Pending TypeDoc limitations, of course)

Reply:
Typedoc doesn't seem to have `@example`.  Could try to split it out though?  Would need a good way of previewing the TypeDoc to check the results.

Reply:
Nevermind, turns out it does have it.